pageserver/src/tenant/storage_layer/inmemory_layer/vectored_dio_read.rs

```diff
@@ -0,0 +1,916 @@
+use std::{
+    collections::BTreeMap,
+    sync::{Arc, RwLock},
+};
+
+use itertools::Itertools;
+use tokio_epoll_uring::{BoundedBuf, IoBufMut, Slice};
+
+use crate::context::RequestContext;
+
+mod sealed {
+    pub trait Sealed {}
+}
+
+/// The file interface we require. At runtime, this is a [`crate::virtual_file::VirtualFile`].
+pub trait File: Send {
+    async fn read_at_to_end<'a, 'b, B: IoBufMut + Send>(
```

Comment:
nit: Perhaps I'm being pedantic, but this name is ambiguous. It can mean read to the end of the file or read
to the end of the buffer.

Reply:
https://github.com/neondatabase/neon/pull/8537/commits/df4571fef17f3a77fa44212503be7aae9831d76f